library/core/src/intrinsics.rs

```diff
@@ -846,6 +846,9 @@ extern "rust-intrinsic" {
     /// destination value, then forgets the original. It's equivalent to C's
     /// `memcpy` under the hood, just like `transmute_copy`.
     ///
+    /// Because `transmute` is a by-value operation, alignment is not a concern.
+    /// The compiler already ensures both `T` and `U` are properly aligned.
```

Comment:
I think I might want to say something like "(but this is not unique to transmute)" or so, since there's nothing really interesting going on here with regards to alignment - transmute is just a function like any other in that regard.

Reply:
I now mention that this is a property that holds for any functions, which implicitly hints that this is not unique to transmute. But I'm happy to change the wording again if you think it's not clear enough.